diff -ur nginx-sticky-module-1.1.orig/ngx_http_sticky_misc.c nginx-sticky-module-1.1/ngx_http_sticky_misc.c
--- nginx-sticky-module-1.1.orig/ngx_http_sticky_misc.c	2012-09-24 09:48:21.000000000 -0700
+++ nginx-sticky-module-1.1/ngx_http_sticky_misc.c	2013-11-08 14:03:41.200018436 -0800
@@ -16,6 +16,16 @@
 	#define ngx_str_set(str, text) (str)->len = sizeof(text) - 1; (str)->data = (u_char *) text
 #endif
 
+static cookie_expires(char *str, size_t size, time_t t)
+{
+	char *months[] = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
+	char *wdays[] = {"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
+	struct tm  e;
+	gmtime_r(&t, &e);
+	return snprintf(str, size, "%s, %02d-%s-%04d %02d:%02d:%02d GMT",
+		wdays[e.tm_wday], e.tm_mday, months[e.tm_mon], e.tm_year + 1900, e.tm_hour,e.tm_min,e.tm_sec);
+}
+
 ngx_int_t ngx_http_sticky_misc_set_cookie(ngx_http_request_t *r, ngx_str_t *name, ngx_str_t *value, ngx_str_t *domain, ngx_str_t *path, time_t expires)
 {
 	u_char  *cookie, *p;
@@ -24,6 +34,8 @@
 	ngx_str_t remove;
 	ngx_list_part_t *part;
 	ngx_uint_t i;
+	char expires_str[80];
+	int expires_len;
 
 	if (value == NULL) {
 		ngx_str_set(&remove, "_remove_");
@@ -38,9 +50,10 @@
 		len += sizeof("; Domain=") - 1 + domain->len;
 	}
 
-	/*; Max-Age= */
+	/*; Expires= */
 	if (expires != NGX_CONF_UNSET) {
-		len += sizeof("; Max-Age=") - 1 + NGX_TIME_T_LEN;
+		expires_len = cookie_expires(expires_str, sizeof(expires_str), time(NULL) + expires);
+		len += sizeof("; Expires=") - 1 + expires_len;
 	}
 
 	/* ; Path= */
@@ -63,8 +76,8 @@
 	}
 
 	if (expires != NGX_CONF_UNSET) {
-		p = ngx_copy(p, "; Max-Age=", sizeof("; Max-Age=") - 1);
-		p = ngx_snprintf(p, NGX_TIME_T_LEN, "%T", expires);
+		p = ngx_copy(p, "; Expires=", sizeof("; Expires=") - 1);
+		p = ngx_copy(p, expires_str, expires_len);
 	}
 
 	if (path->len > 0) {
